src/any_agent/frameworks/smolagents.py

```diff
@@ -71,28 +73,45 @@ async def _load_agent(self) -> None:
             self._agent.prompt_templates["system_prompt"] = self.config.instructions
 
         if self.config.output_type:
-            output_type = self.config.output_type
-
-            class CustomFinalAnswerTool(FinalAnswerTool):  # type: ignore[no-untyped-call]
-                inputs = {  # noqa: RUF012
-                    "answer": {
-                        "type": "string",
-                        "description": f"The final answer to the problem. The input must be a string that conforms to the{output_type.__name__} object.",
+            instructions, final_output_function = prepare_final_output_config_function(
+                self.config.output_type, self.config.instructions
+            )
+
+            # Create a custom tool for smolagents that wraps our final output function
+            class FinalAnswerToolWrapper(FinalAnswerTool):  # type: ignore[no-untyped-call]
```

Comment:
since there's an init and (presumably) the class does not depend on the enclosing scope, could it be moved out of the `_load_agent` definition?

Reply:
It does depend on the enclosing scope, so unfortunately it needs to stay here.